### PR TITLE
Render documentation descriptions as Markdown

### LINKED ISSUE #1337: Allow cross-references to other parts of the documentation
It would be helpful if there were an easy way to cross-reference to other parts of the documentation from within it. For instance, the documentation for [`fill()`](http://p5js.org/reference/#/p5/fill) mentions [`colorMode()`](http://p5js.org/reference/#/p5/colorMode):

colormode

It would be great if that was actually a link to the documentation for `colorMode()` rather than just static text.

YUIDoc does seem to have a mechanism for [cross-referencing modules and classes](http://yui.github.io/yuidoc/syntax/index.html#cross-referencing-modules-and-classes), but as far as I can tell, it assumes a certain syntax for documentation URLs that p5 doesn't conform to. I'm also not a huge fan of how verbose it is.

So, one option is to come up with our own alternative formatting. I guess one way we could do it, which might even work without having to write any additional tooling/preprocessing, is simply to use Markdown to relatively link to the appropriate place, e.g.:

```markdown
depending on the current [`colorMode()`](#/p5/colorMode).
```

I like this approach because it takes advantage of the nice URL fragment syntax the docs already use. It also doesn't really require any explicit documentation, and contributors can easily figure out what the fragment is by just visiting the proper section on the website and copy-pasting the fragment.

If we did things this way, I could probably even add a step to our test suite that checks to make sure that none of these relative links are broken.

Thoughts?

### PR DESCRIPTION
**Pull request by toolness**
While trying to fix #1337, I realized that our documentation descriptions aren't currently rendering as Markdown. p5's use of YUIDoc had led me to assume otherwise.

I believe the reason for this is because YUIDoc is built for *statically* generating documentation, and it renders Markdown during this static generation phase. However, p5's static generation phase is actually extremely simple, because its documentation is a single-page app, rendered at [`main.handlebars`](https://github.com/processing/p5.js/blob/master/docs/yuidoc-p5-theme/layouts/main.handlebars).

Instead, the majority of page rendering is actually done on the client-side via the Backbone app contained in [`docs/yuidoc-p5-theme-src`](https://github.com/processing/p5.js/tree/master/docs/yuidoc-p5-theme-src). This app looks at `data.json` directly to retrieve all the reference documentation data, and *that* file has the original, raw descriptions for our documentation--*not* rendered Markdown.

This is why we have to have `<br><br>`'s between every paragraph in our documentation, rather than simply inserting a blank line, as YUIDoc leads us to believe (which @limzykenneth has noted in #1190).

So this PR attempts to ease this difficulty by using our [YUIDoc preprocessor](https://github.com/processing/p5.js/pull/1295#issuecomment-201275694) to convert all descriptions in `data.json` to Markdown at build time (via the [marked](https://github.com/chjj/marked) npm module). I decided to take this approach rather than rendering the Markdown on the client-side for a few reasons:

* There isn't any dynamic functionality pertaining to Markdown rendering that actually *requires* it to be on the client side.
* Rendering it on the client-side means shipping a Markdown-rendering library to the user's browser, which isn't great given that the [docs already load slowly](https://github.com/processing/p5.js-website/issues/143). (According to GitHub, [`marked.min.js`](https://github.com/chjj/marked/blob/master/marked.min.js) weighs in at 19K.)
  * Actually, in retrospect, there is a tradeoff here: because Markdown is generally more compact than HTML, the `data.json` blob will be larger if we render the Markdown at build time. So not sure if this ends up being a net gain.
* Rendering it on the client-side potentially slows down the client/drains battery life.
* Rendering it at build time means we can (eventually) statically analyze it and ensure its correctness, e.g. that any internal links aren't broken.

That said, because this PR now renders everything as Markdown, it's quite likely that some documentation doesn't appear exactly like it used to.

### CODE REVIEW

File: docs/yuidoc-p5-theme/assets/reference.css [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/**
+ * Note: most of the styling for the reference documentation is actually
+ * contained in the p5.js-website repository:
+ *
+ * https://github.com/processing/p5.js-website/blob/master/css/main.css
+ *
+ * However, some of it is located here so we can easily iterate on it
+ * in the same repository.
```
Comment (toolness) [inline docs/yuidoc-p5-theme/assets/reference.css]: Um, sorry if this is a terrible idea :sweat:

Would it be useful to actually move all the reference-related CSS from [p5.js-website's `main.css`](https://github.com/processing/p5.js-website/blob/master/css/main.css) over here, though? Or is some of that reference CSS used elsewhere on the p5.js website?

**Comment by limzykenneth** [comment]
I'm a little bit confused...with this PR will description in data.json be in markdown or rendered html from markdown?

**Comment by toolness** [comment]
It will be in rendered HTML from markdown. We can change it to contain the original markdown, but this will mean rendering the markdown on the client-side (in the Backbone app), where we will incur the opposite of the tradeoffs I listed in the description above.

Even if we do that, though, we'll likely (eventually) want to at least *parse* the markdown at build time to ensure there aren't any broken links and such.

**Comment by limzykenneth** [comment]
hmm...do you have a branch where the data.json will be in markdown and we parse markdown on client side? I'd like to do some tests and comparisons see what works best.

**Comment by toolness** [comment]
Ah, nope, unfortunately I do not currently have such a branch. :/

File: docs/yuidoc-p5-theme/assets/reference.css (hunk 1 above)
Comment (lmccart) [inline docs/yuidoc-p5-theme/assets/reference.css]: The main.css file holds all the css for the whole site. I suppose we could separate it out, but there are some things we want to ensure keep consistent between the reference and the rest of the site, and it seems like this might just end up in duplicated code?

**Comment by lmccart** [comment]
This all looks good, let's give it a try. Thanks @toolness. @limzykenneth if you do any further testing in this direction let us know, this is all a work very much in progress as you know. :)

**Comment by limzykenneth** [comment]
Cool, will report back as soon as I find some extra time to do some test on this... :sob:
